### PR TITLE
fix url for call to search for layers

### LINKED ISSUE #506: BasedOnUbi fails with Error when querying pyxis for uncompressed top layer idsRecieved http status 404 Not Found
## Bug Description
BasedOnUBI fails with Error when querying pyxis for uncompressed top layer idsRecieved http status 404 Not Found
## Version and Command Invocation
(The output of `preflight --version`)
preflight version 0.0.0 <commit: 5fd339e59c1faa01cb8bf8e72754e6b8660400a3>

## Steps to Reproduce:
(How can we reproduce this?)
1) ./preflight check container registry.connect.redhat.com/haproxytech/haproxy:1.8.23-1624151643   --pyxis-api-token XXX --certification-project-id xxxxx --pyxis-host catalog.xx.redhat.com/api/containers
2)
3)

## Expected Result
BasedOnUBI test should pass for certified images

## Actual Result
```
./preflight check container registry.connect.redhat.com/haproxytech/haproxy:1.8.23-1624151643   --pyxis-api-token XXX --certification-project-id XXX --pyxis-host catalog.qa.redhat.com/api/containers
time="2022-03-21T13:24:13Z" level=debug msg="config file not found, proceeding without it"
time="2022-03-21T13:24:13Z" level=info msg="certification library version 0.0.0 <commit: 5fd339e59c1faa01cb8bf8e72754e6b8660400a3>"
time="2022-03-21T13:24:13Z" level=debug msg="URL is: https://catalog.qa.redhat.com/api/containers/v1/projects/certification/id/XXX"
time="2022-03-21T13:24:16Z" level=debug msg="Certification project name is: Viksss"
time="2022-03-21T13:24:16Z" level=debug msg="target image: registry.connect.redhat.com/haproxytech/haproxy:1.8.23-1624151643"
time="2022-03-21T13:24:16Z" level=debug msg="pulling image from target registry"
time="2022-03-21T13:24:18Z" level=debug msg="temporary directory is /tmp/preflight-1056765935"
time="2022-03-21T13:24:18Z" level=debug msg="exporting and flattening image"
time="2022-03-21T13:24:18Z" level=debug msg="extracting container filesystem to /tmp/preflight-1056765935/fs"
time="2022-03-21T13:24:18Z" level=debug msg="writing container filesystem to output dir: /tmp/preflight-1056765935/fs"
time="2022-03-21T13:29:11Z" level=trace msg="image config written to disk: /home/vagrant/openshift-preflight/artifacts/cert-image.json"
time="2022-03-21T13:29:11Z" level=trace msg="rpm manifest written to disk: /home/vagrant/openshift-preflight/artifacts/rpm-manifest.json"
time="2022-03-21T13:29:11Z" level=debug msg="Container checks do not require a cluster. skipping cluster version check."
time="2022-03-21T13:29:11Z" level=debug msg="executing checks"
time="2022-03-21T13:29:11Z" level=debug msg="running check: LayerCountAcceptable"
time="2022-03-21T13:29:11Z" level=debug msg="detected 6 layers in image"
time="2022-03-21T13:29:11Z" level=info msg="check completed: LayerCountAcceptable" result=PASSED
time="2022-03-21T13:29:11Z" level=debug msg="running check: HasNoProhibitedPackagesMounted"
time="2022-03-21T13:29:11Z" level=info msg="check completed: HasNoProhibitedPackagesMounted" result=PASSED
time="2022-03-21T13:29:11Z" level=debug msg="running check: HasRequiredLabel"
time="2022-03-21T13:29:11Z" level=info msg="check completed: HasRequiredLabel" result=PASSED
time="2022-03-21T13:29:11Z" level=debug msg="running check: RunAsNonRoot"
time="2022-03-21T13:29:11Z" level=debug msg="User specified that was not root"
time="2022-03-21T13:29:11Z" level=info msg="check completed: RunAsNonRoot" result=PASSED
time="2022-03-21T13:29:11Z" level=debug msg="running check: BasedOnUbi"
time="2022-03-21T13:29:11Z" level=trace msg="the layerIds passed to pyxis are [sha256:b15f595a167c0358e426546a93b192e5908728bc8cb37feff1b8c14338ce0b56 sha256:8b4e379ab3b791800538f248cc775f857cf94bfcd233b802949ea16440258ed1 sha256:4ea1d6680dba452c765b8fbd74c5732d8087edc1ed5f7f35143b58928a1000ca sha256:a0462ea721dae205dd2baaa1110eb48031bc1328daec7ce50fcf67afb953e830 sha256:10139ad441e5ef65f3cc2575b6b88466e800461fffb9622b8bf75d06f7975818 sha256:7903d7da767adefd90ac59d5d0d56563cd7be1bd67833497471ff5108ff67334]"
time="2022-03-21T13:29:13Z" level=error msg="Unexpected Status CodeRecieved http status 404 Not Found instead of 200 OK"
time="2022-03-21T13:29:13Z" level=error msg="Error when querying pyxis for uncompressed top layer idsRecieved http status 404 Not Found instead of 200 OK"
time="2022-03-21T13:29:13Z" level=info msg="check completed: BasedOnUbi" result=FAILED
time="2022-03-21T13:29:13Z" level=debug msg="running check: HasModifiedFiles"
time="2022-03-21T13:31:12Z" level=info msg="check completed: HasModifiedFiles" result=PASSED
time="2022-03-21T13:31:12Z" level=debug msg="running check: HasLicense"
time="2022-03-21T13:31:12Z" level=debug msg="3 Licenses found"
time="2022-03-21T13:31:12Z" level=info msg="check completed: HasLicense" result=PASSED
time="2022-03-21T13:31:12Z" level=debug msg="running check: HasUniqueTag"
time="2022-03-21T13:31:14Z" level=info msg="check completed: HasUniqueTag" result=PASSED
{
    "image": "registry.connect.redhat.com/haproxytech/haproxy:1.8.23-1624151643",
    "passed": false,
    "test_library": {
        "name": "github.com/redhat-openshift-ecosystem/openshift-preflight",
        "version": "0.0.0",
        "commit": "5fd339e59c1faa01cb8bf8e72754e6b8660400a3"
    },
    "results": {
        "passed": [
            {
                "name": "LayerCountAcceptable",
                "elapsed_time": 0,
                "description": "Checking if container has less than 40 layers.  Too many layers within the container images can degrade container performance."
            },
            {
                "name": "HasNoProhibitedPackagesMounted",
                "elapsed_time": 103,
                "description": "Checks to ensure that the image in use does not include prohibited packages, such as Red Hat Enterprise Linux (RHEL) kernel packages."
            },
            {
                "name": "HasRequiredLabel",
                "elapsed_time": 0,
                "description": "Checking if the required labels (name, vendor, version, release, summary, description) are present in the container metadata."
            },
            {
                "name": "RunAsNonRoot",
                "elapsed_time": 0,
                "description": "Checking if container runs as the root user because a container that does not specify a non-root user will fail the automatic certification, and will be subject to a manual review before the container can be approved for publication"
            },
            {
                "name": "HasModifiedFiles",
                "elapsed_time": 119647,
                "description": "Checks that no files installed via RPM in the base Red Hat layer have been modified"
            },
            {
                "name": "HasLicense",
                "elapsed_time": 0,
                "description": "Checking if terms and conditions applicable to the software including open source licensing information are present. The license must be at /licenses"
            },
            {
                "name": "HasUniqueTag",
                "elapsed_time": 1473,
                "description": "Checking if container has a tag other than 'latest', so that the image can be uniquely identified."
            }
        ],
        "failed": [
            {
                "name": "BasedOnUbi",
                "elapsed_time": 1384,
                "description": "Checking if the container's base image is based upon the Red Hat Universal Base Image (UBI)",
                "help": "Check BasedOnUbi encountered an error. Please review the preflight.log file for more information.",
                "suggestion": "Change the FROM directive in your Dockerfile or Containerfile to FROM registry.access.redhat.com/ubi8/ubi",
                "knowledgebase_url": "https://connect.redhat.com/zones/containers/container-certification-policy-guide",
                "check_url": "https://connect.redhat.com/zones/containers/container-certification-policy-guide"
            }
        ],
        "errors": []
    }
}


## Additional Context
(Anything else you think might help us troubleshoot, like your platform, dependency versions, etc).

```

### PR DESCRIPTION
**Pull request by skattoju**
fixes #506 
added logs for clarity

### CODE REVIEW

**Review by acornett21 — CHANGES_REQUESTED** [review]
Body is a byte array so I think it needs to be converted to use`%s`

File: certification/pyxis/layers.go [hunk 1]
```diff
@@ -43,15 +45,15 @@ func (p *pyxisEngine) CheckRedHatLayers(ctx context.Context, layerHashes []crane
...
-
 	defer resp.Body.Close()
-
 	body, err := io.ReadAll(resp.Body)
 	if err != nil {
 		log.Error(err)
 		return nil, err
 	}
 
+	log.Tracef("query response from pyxis %s", body)
```
[the remark this answers is not in the document]
Comment (skattoju) [reply certification/pyxis/layers.go]: > Body is a byte array so I think it needs to be converted to use`%s`

yeah looks weird in the logs its string now 👍

**Review by bcrochet — APPROVED** [review]
/lgtm

**Pull request merged**